ecs-agent/wsclient/client.go

```diff
@@ -0,0 +1,577 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Package wsclient wraps the generated aws-sdk-go client to provide marshalling
+// and unmarshalling of data over a websocket connection in the format expected
+// by backend. It allows for bidirectional communication and acts as both a
+// client-and-server in terms of requests, but only as a client in terms of
+// connecting.
+package wsclient
```

Comment:
Periodic timeout/disconnection of WebSocket connections seems to be missing from this PR, let’s make sure to add that in if we can.

Reply:
Currently, the timeouts are initiated by individual components using the library, i.e. ACS and TACS have their own implementation of timeouts. I had not accounted for that as part of wsclient library, hence it's missing from this PR. Is that something we want to unify?

Reply:
Yeah, I believe we decided that we want to have the same logic for refreshing our websocket connection across all our websocket connections (i.e., ACS and TACS websocket connections). I'll reach out to you offline and point you in the direction of these previous discussions.